modules/signals/src/ts-helpers.ts

```diff
@@ -27,3 +27,11 @@ export type IsKnownRecord<T> = IsRecord<T> extends true
 export type OmitPrivate<T> = {
   [K in keyof T as K extends `_${string}` ? never : K]: T[K];
 };
+
+export type DeepReadonly<T> = T extends object
```

Comment:
The [`type-fest`](https://github.com/sindresorhus/type-fest) library has a more comprehensive version of this: https://github.com/sindresorhus/type-fest/blob/main/source/readonly-deep.d.ts. I can't tell the differences off the top of my head but thought I'd mention it.

(Note: I'm not advocating using the `type-fest` library as a dependency — more providing this in case of inspiration for some good 'ol copy-pasting :))

Reply:
Hey @jits, I took a look at it before. It supports a lot of types that are not allowed to be part of the state object. So I took arrays, primitive types and nested objects out of it.

Reply:
Hi @rainerhahnekamp — ahh, perfect! Thanks for clarifying. (And that makes a lot of sense)